text/0064-publishing-protobufs.md

```diff
@@ -0,0 +1,198 @@
+    │       └── lib.rs
+    ├── tonic
+    │   ├── build.rs
+    │   ├── Cargo.toml
+    │   ├── protos
+    │   │   └── services.proto (Symlink to <version>/services.proto)
+    │   ├── README.md
+    │   └── src
+    │       └── lib.rs
+    └── <version> (of the form `v1`, `v2`, etc.)
+        ├── messages.proto
+        └── services.proto
+```
+
+The `<name of api>` is used to separate different Protobuf APIs. For example
+this might be `attestatoin` to define only the attestation Protobuf API.
+
+The `grpcio` directory holds the files to build the rust crate
+`mc-<name-of-api>-grpcio`. This rust crate is for building the services that
+work with [rust grpcio](https://docs.rs/grpcio/latest/grpcio/).
+
+The `messages` directory holds the files to build the rust crate
+`mc-<name-of-api>-messages`. This rust crate is for building the
+[prost](https://docs.rs/prost/latest/prost/) messages. 
+
+The `tonic` directory holds the files to build the rust crate
+`mc-<name-of-api>-tonic`. This rust crate is for building the services that work
+with [tonic](https://docs.rs/tonic/latest/tonic/). 
+
+The `<version>` directory is used to isolate breaking changes to the Protobuf
+API. All updates to `v1` should continue to work with the first published
+release of `v1`. If a breaking change is necessary it should be made in a
+subsequent `v2` directory.
+
+Symlinks are used from the rust crates back to the `<version>` directory to
+ensure that the rust crates build on <https://crates.io>.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Using the Buf Schema Registry puts reliance on a third party. It is currently
+  listed as "in beta", meaning it may have changes coming that we may need to
+  account for as time progresses. 
+
+- Housing the rust crates in the Protobuf repo may encourage housing other
+  language versions in the repo.
+
+- Separaing the Protobuf definitions from the main repo can make it less
+  ergonomic to develop newer APIs. Developers may need to temporarily redirect
+  crate dependencies to current inflight API changes.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Publishing the Protobuf API definitions makes it easier for other code bases to
+communicate with Consensus and Fog. Regardless of which language is used,
+clients can generate code to communicate using the API without having to find
+the files in the MobileCoin Foundation repo.
+
+The Protobuf APIs can be versioned independent of the Consensus and Fog
+releases. This makes it easier for a client to see when pertenent changes
+occur.
+
+Having a dedicated repo makes it easier to have CI checks to prevent breaking
+changes to the Protobuf APIs.  Currently it is left to the developers to ensure
+changes are non breaking. The main MobileCoin Foundation repo already has quite
+a lot of CI steps adding another step for Protobuf breaking changes is likely to
+fall to the wayside.
+
+# Prior art
+[prior-art]: #prior-art
+
+None
```

Comment:
Up for suggestions here, I'm guessing there probably are some practices regarding protobuf publishing but I didn't find much doing a quick google. Probably looking for wrong terms